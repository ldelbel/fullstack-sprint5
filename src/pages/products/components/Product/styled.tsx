import styled from "styled-components";

export const StyledProduct = styled.div`
  @media (max-width: 1200px) {
    flex-basis: 50%;
  }

  @media (min-width: 1201px) {
    flex-basis: 25%;
  }

  .card {
    cursor: pointer;
    border: 1px solid white;
    padding: 8px;
  }

  .card:hover {
    box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),
      0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  }

  .card__img {
    max-width: 100%;
    max-height: 100%;
  }

  .card__description {
    font-size: 0.6875rem;
    font-weight: 400;
    color: rgba(0, 0, 0, 0.87);
    text-transform: uppercase;
    text-align: center;
    margin-top: 8px;
    margin-bottom: 10px;
  }

  .card__price {
    font-size: 0.8125rem;
    font-weight: 600;
    text-transform: uppercase;
    text-align: center;
  }
`;
