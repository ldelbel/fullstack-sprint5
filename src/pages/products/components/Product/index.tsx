import { ReactElement } from "react";
import { Link } from "react-router-dom";
import { TProduct } from "../../../../types";
import { StyledProduct } from "./styled";

interface IProduct {
  product: TProduct;
}

export function Product({ product }: IProduct): ReactElement<React.FC> {
  const { image, name, price } = product;
  const slug = name.toLowerCase().split(" ").join("-");
  const newTo = { pathname: `/${slug}`, state: product };

  return (
    <StyledProduct>
      <Link to={newTo}>
        <div className="card">
          <img className="card__img" src={image} alt="" />
          <p className="card__description">{name}</p>
          <p className="card__price">R$ {price}</p>
        </div>
      </Link>
    </StyledProduct>
  );
}
