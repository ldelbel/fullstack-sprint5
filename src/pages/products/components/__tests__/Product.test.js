import React from "react";
import { render } from "@testing-library/react";
import { Product } from "../Product";
import { MemoryRouter } from "react-router-dom";

const product = {
  sku: 1,
  image: "assets/c1.webp",
  name: "Jaqueta Puffer Juvenil Com Capuz Super Mario Branco Tam 10 a 18",
  price: "199,90",
};

describe("Product component", () => {

  it("should render filters", () => {
    const { getByRole } = render(
      <MemoryRouter>
        <Product product={product} />
      </MemoryRouter>
    );
    expect(getByRole("link")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <Product product={product} />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

});
