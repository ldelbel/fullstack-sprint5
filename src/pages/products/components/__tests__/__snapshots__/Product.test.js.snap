// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component should match snapshot 1`] = `
<div
  class="sc-bdnxRM eDZwqH"
>
  <a
    href="/jaqueta-puffer-juvenil-com-capuz-super-mario-branco-tam-10-a-18"
  >
    <div
      class="card"
    >
      <img
        alt=""
        class="card__img"
        src="assets/c1.webp"
      />
      <p
        class="card__description"
      >
        Jaqueta Puffer Juvenil Com Capuz Super Mario Branco Tam 10 a 18
      </p>
      <p
        class="card__price"
      >
        R$ 
        199,90
      </p>
    </div>
  </a>
</div>
`;
